pkg/template_generator.go

```diff
@@ -130,6 +130,19 @@ func generateTemplateForPodSpec(podSpec kapi.PodSpec, key string, value map[stri
 		podSpec.RestartPolicy = kapi.RestartPolicy(fmt.Sprintf("{{.Values.%s.%s}}", key, RestartPolicy))
 	}
 
+	if len(podSpec.ImagePullSecrets) != 0 {
+		imagePullSecretsObj := []kapi.LocalObjectReference{}
+
+		for _, imagePullSecrets := range podSpec.ImagePullSecrets {
+			imagePullSecrets.Name = generateSafeKey(imagePullSecrets.Name)
+			value["imagePullSecrets"] = imagePullSecrets.Name
```

Comment:
This is an issue we discussed internally. It is not quite clear what is the best option. There are scenarios where users want to use `ImagePullSecrets` in the Pod/Deployment spec but do not want to include that in the chart. So, we can't assume that the `Secret` yaml will be included in the chart.

I think we can just add that as a `Values.imagePullSecret` and update the specs accordingly. It will be up to the Chart authir to decide the proper approach.

Also, how do you want to handle multiple image pull secrets (it's an array) ?